test/uk/gov/hmrc/gform/gform/VariablesBuilderTest.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.gform
+
+import uk.gov.hmrc.auth.core.retrieve.OneTimeLogin
+import uk.gov.hmrc.auth.core.{ Enrolment, EnrolmentIdentifier, Enrolments, AffinityGroup => CoreAffinityGroup }
+import uk.gov.hmrc.gform.Spec
+import uk.gov.hmrc.gform.auth.models.AuthenticatedRetrievals
+import uk.gov.hmrc.gform.graph.processor.IdentifierExtractor
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.generators.FormTemplateGen
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.{ AllowAnyAgentAffinityUser, BasicText, ComponentType, EnrolmentAuth, FormComponent, HmrcAgentWithEnrolmentModule, Never, ServiceId, Text, UserCtx }
+import uk.gov.hmrc.gform.sharedmodel.VariablesBuilder._
+import com.softwaremill.quicklens._
+import uk.gov.hmrc.gform.sharedmodel.VariablesBuilder
+
+class VariablesBuilderTest extends Spec with FormTemplateGen {
+
+  forAll(formTemplateGen) { template =>
+    it should s"Build a data structure with valid key value pair for $template" in new IdentifierExtractor {
```

Comment:
Dumping generated `template` into test description doesn't seems useful at all. Cannot we just dump `template. _id`?

Reply:
Sure